modules/cli/src/main/scala/scala/cli/ScalaCli.scala

```diff
@@ -138,6 +140,17 @@ object ScalaCli extends CommandsEntryPoint {
   }
 
   private def main0(args: Array[String]): Unit = {
+    LauncherOptions.parser.stopAtFirstUnrecognized.parse(args) match {
+      case Left(e) =>
+        System.err.println(e.message)
+        sys.exit(1)
+      case Right((launcherOpts, args0)) =>
+        launcherOpts.cliVersion.map(_.trim).filter(_.nonEmpty) match {
+          case Some(ver) =>
+            LauncherCLI.runAndExit(ver, launcherOpts, args0)
+          case None =>
```

Comment:
I think we should really ensure we use `args0` down below. Users might inadvertently specify `--cli-scala-version …` for example, and we should ignore it if that's the case.